Fix issue 2, and numerous bugs

bos opened the pull request:
- Results of memory allocations were unchecked.
- Neither endianness nor size of header were specified, and could
  thus vary incompatibly across architecture (the Java JNI bindings
  make the same mistake). They are now defined to be 32-bit
  little-endian.
- Potentially malformed (i.e. negative) input to uncompress was
  unchecked.

In addition, we no longer unconditionally resize the result unless
it's more than 25% smaller than the string we allocated.

Commit:
Fix several correctness problems

* Results of memory allocations were unchecked.

* Neither endianness nor size of header were specified, and could
  thus vary incompatibly across architecture (the Java JNI bindings
  make the same mistake). They are now defined to be 32-bit
  little-endian.

* Potentially malformed (i.e. negative) input to uncompress was
  unchecked.

In addition, we no longer unconditionally resize the result unless
it's more than 25% smaller than the string we allocated.

steeve commented:
Fuck yeah! I'll merge it ASAP (probably today or tomorrow).

Thanks a lot!

On 5 juin 2012, at 08:00, Bryan O'Sullivan
reply@reply.github.com
wrote:

> - Results of memory allocations were unchecked.
> - Neither endianness nor size of header were specified, and could
>   thus vary incompatibly across architecture (the Java JNI bindings
>   make the same mistake). They are now defined to be 32-bit
>   little-endian.
> - Potentially malformed (i.e. negative) input to uncompress was
>   unchecked.
> 
> In addition, we no longer unconditionally resize the result unless
> it's more than 25% smaller than the string we allocated.
> 
> You can merge this Pull Request by running:
> 
>  git pull https://github.com/bos/python-lz4 master
> 
> Or you can view, comment on it, or merge it online at:
> 
>  https://github.com/steeve/python-lz4/pull/4
> 
> -- Commit Summary --
> - Fix several correctness problems
> 
> -- File Changes --
> 
> M src/python-lz4.c (84)
> 
> -- Patch Links --
> 
>  https://github.com/steeve/python-lz4/pull/4.patch
>  https://github.com/steeve/python-lz4/pull/4.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/steeve/python-lz4/pull/4

Commit:
Update to latest lz4 sources (r66)

Commit:
Add support for compressHC

This gives compression ratios close to zlib, but with performance close to zlib
too. Decompression speed is about 5% better than with plain lz4 compressed
data (~1GB/sec).

steeve commented:
Awesome fixes, optims !

steeve commented:
0.4.0 published on PyPI: http://pypi.python.org/pypi/lz4
